src/lsqecc/pauli_rotations/segmented_qasm_parser.py

```diff
@@ -82,16 +82,22 @@ def segment_to_circuit(self, segment: Segment):
     def reversible_segment_to_circuit(
         self, segment: List[qiskit.qasm.node.node.Node]
     ) -> PauliOpCircuit:
-        program_wrapper_node = qiskit.qasm.node.Program(segment)
+        gates_as_qasm = qiskit.qasm.node.Program(segment).qasm()
+
         text_qasm_program_wrapper = (
             "OPENQASM 2.0;\n"
             + 'include "qelib1.inc";\n'
             + self.qreg.qasm()
             + "\n"
-            + program_wrapper_node.qasm()
+            + gates_as_qasm
         )
+        # Feeding text_qasm_program_wrapper directly to pyzx qould cause it to complain because
+        # gates_as_qasm, being generated directly from the AST, has extra parentheses pyzx doesn't
+        # like. So we feeding it back to qiskit we remove those extra parentheses
+        # TODO try avoid this step by going directly to circuit
```

Comment:
Probably not worth addressing this TODO because, in light of #233 we might have to drop PyZX